java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -244,7 +244,7 @@ class DatabaseImpl(
             dbSingletons.forEach { (fieldName, value) -> rawSingletons[fieldName] = value }
             dbCollections.forEach { (fieldName, value) -> rawCollections[fieldName] = value }
 
-            return DatabaseData.Entity(
+            return@forSingleResult DatabaseData.Entity(
```

Comment:
Is there a test we can add that would have caught this?

Reply:
I saw @jblebrun had same question. If adding instrumentation tests for this class (or other logic in this package) would be helpful to catch this and other sorts of issues we've been turning up, perhaps worth filing an issue and referencing this PR as an example to kick start such tests? @yuangu-google